File: APP/src/app/models/Task.ts
export class Task {
  id: number;
  categoryID: number;  
  userID: string;
  title: string;
  description?: string;
  dueDate?: Date;
  isCompleted: boolean;
  priorityID: number;
  createdAt: Date;
  updatedAt: Date;
  priority?: Priority;
  user?: AppUser;
  category?: Category;

  constructor() {
    this.id = 0;
    this.categoryID = 0;
    this.userID = '';
    this.title = '';
    this.isCompleted = false;
    this.priorityID = 0;
    this.createdAt = new Date();
    this.updatedAt = new Date();
  }
}


export class Priority {
  id: number;
  name: string;
  
  constructor() {
    this.id = 0;
    this.name = '';
  }
}

export class AppUser {
  id: string;
  constructor() {
    this.id = '';
  }
}

export class Category {
  id: number;
  name: string;
  
  constructor() {
    this.id = 0;
    this.name = '';
  }
}

File: DotNet_API/Entities/Task.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Diagnostics;

namespace DotNet_API.Entities
{

   
    public class Task
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ID { get; set; }

        [Required]
        public int CategoryID { get; set; }

        [Required]
        public string UserID { get; set; } = string.Empty;

        [Required]
        [StringLength(50)]
        public string Title { get; set; } = string.Empty;

        [StringLength(255)]
        public string? Description { get; set; }

        public DateTime? DueDate { get; set; }

        public bool IsCompleted { get; set; } = false;

        [Required]
        public int PriorityID { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

      




    }

 


}
